### ISSUE TITLE
[Feature🚀] Implement ControllerElectMaster Request Handler

### ISSUE THREAD

**Issue by mxsm** [mxsm/rocketmq-rust#5538]
# Feature Request: Implement ControllerElectMaster Request Handler

## Feature Description

Implement the `handle_elect_master` method in `ControllerRequestProcessor` to handle master election requests for broker groups. This feature is part of the controller's core functionality for managing broker high availability through master failover.

## Problem/Motivation

Currently, the `handle_elect_master` method in the `ControllerRequestProcessor` is unimplemented (returns `unimplemented!` macro). When a broker master fails or during planned maintenance, the system needs to elect a new master from the available replicas. Without this implementation, the controller cannot perform automatic master election, making the system unable to handle master broker failures.

This request is triggered by `RequestCode::ControllerElectMaster` and needs to be handled to ensure high availability of the RocketMQ cluster.

### Location
- **File**: `rocketmq-controller/src/processor/controller_request_processor.rs`
- **Method**: `handle_elect_master` (lines 311-323)
- **Request Code**: `RequestCode::ControllerElectMaster`

### Implementation Steps

1. **Decode Request Header**:
   - Create/use `ElectMasterRequestHeader` in `rocketmq-remoting/protocol/header/controller/`
   - Decode request header containing:
     - `broker_name`: The broker group name
     - `cluster_name`: The cluster name
     - Additional parameters for election criteria

2. **Forward to Controller**:
   - Call `self.controller_manager.controller().elect_master(&request_header).await`
   - The Controller layer should handle:
     - Leader state validation (only leader can process)
     - Raft consensus proposal submission
     - State machine application via ReplicasInfoManager

3. **Return Response**:
   - Return `RemotingCommand` response with:
     - New master broker information (broker ID, address)
     - Success/failure status
     - Error message if election fails

### Example Implementation Pattern

```rust
async fn handle_elect_master(
    &mut self,
    _channel: Channel,
    _ctx: ConnectionHandlerContext,
    request: &mut RemotingCommand,
) -> RocketMQResult<Option<RemotingCommand>> {
    use rocketmq_error::RocketMQError;
    use rocketmq_remoting::protocol::header::controller::elect_master_request_header::ElectMasterRequestHeader;

    // Decode request header
    let request_header = request
        .decode_command_custom_header::<ElectMasterRequestHeader>()
        .map_err(|e| {
            RocketMQError::request_header_error(format!(
                "Failed to decode ElectMasterRequestHeader: {:?}", e
            ))
        })?;

    // Forward to controller for Raft consensus and election logic
    self.controller_manager
        .controller()
        .elect_master(&request_header)
        .await
}
```

## Alternatives

- Could implement client-side master election, but that would be less reliable and require coordination across all clients
- Could use ZooKeeper for election, but the current architecture uses Raft for consistency

### Dependencies Required

1. **Request Header Type**:
   - Create `ElectMasterRequestHeader` in `rocketmq-remoting/src/protocol/header/controller/`
   - Implement `CommandCustomHeader` trait

2. **Controller Method**:
   - Implement `elect_master(&self, header: &ElectMasterRequestHeader) -> RocketMQResult<Option<RemotingCommand>>` in Controller trait
   - Implement in both `RaftController` and `DLedgerController`

3. **State Management**:
   - Use `ReplicasInfoManager` to track replica states
   - Use broker heartbeat data to determine eligible replicas
   - Consider sync state data (ISR - In-Sync Replicas)

### Related Code

- Similar pattern to `handle_alter_sync_state_set` (lines 221-299)
- Uses `WAIT_TIMEOUT_SECONDS` for async operations
- Requires Raft consensus for state changes

### References

- Apache RocketMQ Java implementation: `ElectMasterRequestHeader` and corresponding processor
- Current file: [controller_request_processor.rs](rocketmq-controller/src/processor/controller_request_processor.rs#L311-L323)

**Comment by aditya-rajpurohit** [comment]
@mxsm i'd like to work on this issue!
